src/core/document.js

```diff
@@ -258,6 +258,22 @@ class Page {
     );
   }
 
+  async replaceIdByRef(annotations, deletedAnnotations) {
+    let idToRefMap = null;
+    for (const annotation of annotations) {
+      if (annotation.id) {
+        idToRefMap ||= await this._idToRefMap;
+        const ref = idToRefMap.get(annotation.id);
```

Comment:
Don't you need to handle a `ref` not being found here, to prevent issues below in this method (since calling `RefSet.property.put` with `undefined` will throw)?
(As mentioned previously, in some *very rare* cases annotations may be "raw" dictionaries.)
```suggestion
        const ref = idToRefMap.get(annotation.id);
        if (!ref) {
          continue;
        }
```

Reply:
Do you have an example of a pdf containing such annotations ?